### ISSUE TITLE
Global Styles: Focus outline of block items is missing in Blocks menu

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#53600]
If you use the keyboard to focus on a block in the Blocks menu of the Global Styles menu, the outline will extend the full width of the content and the left and right sides of the outline will not be visible.

![before](https://github.com/WordPress/gutenberg/assets/54422211/086511ff-771d-4969-9d68-39a0e34b486e)

My sense is that you would expect the outline to look something like this:

![after](https://github.com/WordPress/gutenberg/assets/54422211/e85b074e-e17a-4b71-97b1-7383aa5e1d97)

**Comment by josephfusco** [comment]
I can confirm this is an issue.

This PR seems related: https://github.com/WordPress/gutenberg/pull/51787

I should have a PR for this at some point today.

**Comment by josephfusco** [comment]
[PR ready](https://github.com/WordPress/gutenberg/pull/53929)
